pkgs/applications/misc/far2l/default.nix

```diff
@@ -1,29 +1,23 @@
 { lib, stdenv, fetchFromGitHub, fetchpatch, makeWrapper, cmake, pkg-config, wxGTK30, glib, pcre, m4, bash,
-  xdg-utils, gvfs, zip, unzip, gzip, bzip2, gnutar, p7zip, xz, imagemagick, darwin }:
+  xdg-utils, gvfs, zip, unzip, gzip, bzip2, gnutar, p7zip, xz, imagemagick, Cocoa?[],
+  libuchardet, spdlog, xercesc, fmt, openssl, libssh, samba, neon, libnfs, libarchive }:
```

Comment:
```suggestion
, xdg-utils, gvfs, zip, unzip, gzip, bzip2, gnutar, p7zip, xz, imagemagick, Cocoa
, libuchardet, spdlog, xercesc, fmt, openssl, libssh, samba, neon, libnfs, libarchive }:
```

Reply:
I can't change `Cocoa?[]` to `Cocoa` because it won't build on my system then. _Anonymous function called without required argument 'Cocoa'._ I guess you gotta have darwin for that to work.

I have no idea how codebase is supposed to compile against darwin. I suppose that means MacOS? I don't have a Mac to test it, so I'm locked out on this. For now the main goal for me is to make far2l runnable for regular GUI-less NixOS at least. So, speaking of darwin, probably leaving it exactly as it were before for now, or marking as broken (how specifically?) would be the best option.

Reply:
> I can't change Cocoa?[] to Cocoa because it won't build on my system then. Anonymous function called without required argument 'Cocoa'. I guess you gotta have darwin for that to work.

You need to specify it like 

``` 
748   aminal = callPackage ../applications/terminal-emulators/aminal {
749   | inherit (darwin.apple_sdk.frameworks) Carbon Cocoa Kernel;                                                                                                                          750   };
```

in top-level.

> I have no idea how codebase is supposed to compile against darwin. 

I tried building it against darwin and it is broken. I would suggest to remove all traces of Cocoa and add my next suggestion and then we can finally merge.

Reply:
OK.

Based on what I know about Nix, it should be possible to spawn a nix-shell environment which mocks itself as a proper Darwin and try building in that mode on my machine, is this so?